scaladoc/src/dotty/tools/scaladoc/Scaladoc.scala

```diff
@@ -100,7 +100,15 @@ object Scaladoc:
       }
 
     val commonScalaSettings = (new SettingGroup with CommonScalaSettings).allSettings
-    allSettings.filter(commonScalaSettings.contains).foreach(setInGlobal)
+    val allScalaSettings = (new SettingGroup with AllScalaSettings).allSettings
+
+    val (shared, other) = allSettings
+      .filter(s => !s.isDefaultIn(summary.sstate))
+      .filter(allScalaSettings.contains)
+      .partition(commonScalaSettings.contains)
+    shared.foreach(setInGlobal)
+
+    if !other.isEmpty then report.echo(s"Skipping unused scalacOptions: ${other.map(_.name).mkString(", ")}")
```

Comment:
If sbt automatically passes these options to scaladoc, then we shouldn't display this message to the user since they can't do anything about it, so it would just be noise.

Reply:
If we could distinguish which options were user-defined and which were just copied from Compile / scalacOptions that would be nice indeed, but as it is I'm not sure.